### PR TITLE
Fix some helm gen bugs

### PR DESCRIPTION
**Pull request by josh-pritchard**
Found some bugs in the schema generation and name override was not implemented for docs or schema.json

BOT NOTES: 
resolves https://github.com/solo-io/skv2/issues/434

### CODE REVIEW

File: codegen/render/funcs.go [hunk 1]
```diff
@@ -929,20 +921,37 @@ func mergeJsonSchema(
...
+					target.Properties.Set(prop, targetValSchema)
+					merged = true
+				}
+			}
+		}
+
+		if !merged {
+			target.Properties.Set(prop, sourceVal)
+		}
+	}
```
Comment (josh-pritchard) [inline codegen/render/funcs.go:954]: This was a bug in schema gen, merge was was not recursive

File: codegen/render/funcs.go [hunk 2]
```diff
@@ -963,11 +972,11 @@ func jsonSchemaForOperator(
 	}
 
 	// nest the operator values schema in the correct place
-	path := []string{strcase.ToLowerCamel(o.Name)}
+	path := []string{o.FormattedName()}
 	if o.ValuePath != "" {
 		splitPath := strings.Split(o.ValuePath, ".")
 		reverse(splitPath)
-		path = append(splitPath, path...)
+		path = append(path, splitPath...)
```
Comment (josh-pritchard) [inline codegen/render/funcs.go:979]: This was backwards

File: codegen/model/chart.go [hunk 3]
```diff
@@ -215,7 +223,7 @@ func (c Chart) GenerateHelmDoc() string {
 
 	// generate documentation for operator values
 	for _, operatorWithValues := range helmValues.Operators {
-		name := strcase.ToLowerCamel(operatorWithValues.Name)
+		name := operatorWithValues.FormattedName()
```
Comment (josh-pritchard) [inline codegen/model/chart.go:226]: change to make override work in doc generation

File: codegen/render/funcs.go [hunk 4]
```diff
@@ -963,11 +972,11 @@ func jsonSchemaForOperator(
 	}
 
 	// nest the operator values schema in the correct place
-	path := []string{strcase.ToLowerCamel(o.Name)}
+	path := []string{o.FormattedName()}
```
Comment (josh-pritchard) [inline codegen/render/funcs.go:975]: change to make override work in schema.json

File: codegen/render/funcs.go [hunk 5]
```diff
@@ -929,20 +921,37 @@ func mergeJsonSchema(
...
+	target.Required = append(target.Required, src.Required...)
+
+	if target.Properties == nil {
+		target.Properties = orderedmap.New()
 	}
 
-	target.Required = append(target.Required, src.Required...)
+	for _, prop := range src.Properties.Keys() {
+		sourceVal, _ := src.Properties.Get(prop)
+		sourceValSchema, sourcIsSchema := sourceVal.(*jsonschema.Schema)
```
Comment (puertomontt) [inline codegen/render/funcs.go:937]: `sourcIsSchema` -> sourceIsSchema

**Review by albertlockett — APPROVED** [review]

**Pull request merged**